src/js/util/exportservice.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
+ *  
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var NodeDomain = require("generator-connection").NodeDomain,
+        Promise = require("bluebird"),
+        log = require("js/util/log");
+
+    /**
+     * The name of the domain to connect to, which corresponds to the name of the generator plugin
+     * @const
+     * @type {string}
+     */
+    var GENERATOR_DOMAIN_NAME = "generator-spaces";
+
+    /**
+     * The port on which the generator websocket server is started.
+     * TODO - This will eventually be fetched from photoshop dynamically
+     * @const
+     * @type {number}
+     */
+    var GENERATOR_PORT = 59596;
+
+    /**
+     * Duration after which we should give up trying to connect to the export plugin
+     * @const
+     * @type {number}
+     */
+    var CONNECTION_TIMEOUT_MS = 2000;
```

Comment:
What do you think about my removing the timeout completely from here, and simply relying on NodeConnection to timeout?  There's a fine line between belt&suspenders and competing/confusing safeguards.